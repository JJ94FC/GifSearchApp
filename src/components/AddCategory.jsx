import { Button, Grid, TextField } from "@mui/material";
import { Box } from "@mui/system";
import React, { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) {
      return;
    }
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <Grid item xs={12}>
      <Box component="form" onSubmit={onSubmit}>
        <Grid container spacing={1}>
          <Grid item xs={9}>
            <TextField
              fullWidth={true}
              id="filled-basic"
              label="Theme"
              variant="filled"
              placeholder="Add a new theme"
              value={inputValue}
              onChange={onInputChange}
            />
          </Grid>
          <Grid item xs={3}>
            <Button
              sx={{ height: "3.5rem" }}
              type="submit"
              variant="contained"
              onClick={onSubmit}
            >
              Add new theme
            </Button>
          </Grid>
        </Grid>
      </Box>
    </Grid>
  );
};
