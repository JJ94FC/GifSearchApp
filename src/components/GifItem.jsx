import React from "react";
import { Box } from "@mui/material";
import PropTypes from 'prop-types'

export const GifItem = ({ title, url }) => {
  return (
    <Box component={'div'} className="gifItem animate__animated animate__fadeInUp">
      <p>
        {title}
      </p>
      <img
        src={url}
        alt={title}
      />
    </Box>
  );
};

GifItem.prototype = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}