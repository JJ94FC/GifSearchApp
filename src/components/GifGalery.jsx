import { Box, Grid, Typography } from "@mui/material";
import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGalery = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  /* QUEDA HACER LA GALERIA EN COLUMNAS */

  /* IMPLEMENTAR REACT QUERY */

  return (
    <Grid item xs={12}>
      <Grid container spacing={1}>
        <Grid item xs={12}>
          <Typography variant="h5" textAlign={"center"}>
            {isLoading ? "Loading..." : category}
          </Typography>
        </Grid>
        <Grid item xs={12}>
          <Box component={'div'} className="gifGridContainer">
            {images.map((img) => (
              <GifItem title={img.title} url={img.url} key={img.id} />
            ))}
          </Box>
        </Grid>
      </Grid>
    </Grid>
  );
};
