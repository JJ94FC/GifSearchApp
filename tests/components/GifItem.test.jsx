import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Testings GifItem file", () => {
  const title = "title test";
  const url = "https://test.com/test.jpg";

  test("should match with the snapshot proptype", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should test that the img have same url and title given", () => {
    render(<GifItem title={title} url={url} />);
    const {src, alt} = screen.getByRole("img");
		expect(src).toBe(url);
		expect(alt).toBe(title);
  });
});
