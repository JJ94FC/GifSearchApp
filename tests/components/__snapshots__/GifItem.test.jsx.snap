// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testings GifItem file should match with the snapshot proptype 1`] = `
<div>
  <div
    class="gifItem animate__animated animate__fadeInUp MuiBox-root css-0"
  >
    <p>
      title test
    </p>
    <img
      alt="title test"
      src="https://test.com/test.jpg"
    />
  </div>
</div>
`;
